### PR TITLE
Pass options along when creating a span with StartSpanFromContext

### PR DESCRIPTION
**Pull request by gwik**
Options were ignored when creating a span.

### CODE REVIEW

**Comment by yurishkuro** [comment]
Do you mind adding unit tests?

**Comment by gwik** [comment]
here you go.

File: testtracer_test.go [hunk 1]
```diff
@@ -21,6 +21,7 @@ func nextFakeID() int {
 type testSpanContext struct {
 	HasParent bool
 	FakeID    int
+	Options   *StartSpanOptions
```
Comment (bensigelman) [inline testtracer_test.go]: I'd have a slight preference to just add tags to `testSpan` and test that they get passed through correctly... if you don't feel like implementing this, I can do it :)

File: testtracer_test.go [hunk 2]
```diff
@@ -28,6 +29,35 @@ func (n testSpanContext) ForeachBaggageItem(handler func(k, v string) bool) {}
 type testSpan struct {
 	spanContext   testSpanContext
 	OperationName string
+	StartTime     time.Time
+	Tags          map[string]interface{}
```
Comment (gwik) [inline testtracer_test.go]: struct equality won't work with a map member, so I implemented an Equal method.

Comment (gwik) [inline testtracer_test.go]: maybe adding an id and compare it would make more sense.

Comment (bensigelman) [inline testtracer_test.go]: LGTM, thanks.

**Comment by bensigelman** [comment]
Thanks @gwik... the test LGTM but I will give @yurishkuro a chance to provide feedback if he so desires...

**Pull request merged**